python/cucim/src/cucim/skimage/filters/tests/test_median.py

```diff
@@ -41,12 +56,167 @@ def test_selem_kwarg_deprecation(image):
+    if out == 'array':
+        out = cp.zeros_like(cam2)
+    assert_allclose(
+        median(cam2, footprint, mode=mode, behavior=behavior, out=out),
+        func(cam2, size=footprint.shape, mode=mode, output=out),
+    )
+
+
+@pytest.mark.parametrize(
+    'mode', ['reflect', 'mirror', 'nearest', 'constant', 'wrap']
+)
+# use an anisotropic footprint large enough to trigger the histogram-based path
+@pytest.mark.parametrize('footprint_shape', [(3, 3), (3, 5), (15, 23)])
+@pytest.mark.parametrize(
+    'int_dtype', [cp.uint8, cp.int8, cp.uint16, cp.int16]
+)
+@pytest.mark.parametrize(
+    'algorithm', ['auto', 'histogram', 'sorting']
+)
+@pytest.mark.parametrize(
+    'algorithm_kwargs', [{}, {'partitions': 32}]
+)
+def test_median_hist_dtypes(
+    mode, footprint_shape, int_dtype, algorithm, algorithm_kwargs,
+):
+    footprint = cp.ones(footprint_shape, dtype=bool)
+    rng = cp.random.default_rng(123)
+    shape = (350, 407)
+    if int_dtype == cp.uint8:
+        img = rng.integers(0, 256, shape, dtype=cp.uint8)
+    elif int_dtype == cp.int8:
+        img = rng.integers(-128, 128, shape, dtype=int).astype(cp.int8)
+    elif int_dtype == cp.uint16:
+        if False:
+            # test with 12-bit range stored in 16-bit integers (e.g. DICOM)
+            img = rng.integers(0, 4096, shape, dtype=cp.uint16)
+        else:
+            # smaller dynamic range
+            #    (range 4096 fails only on CI, but couldn't reproduce locally)
+            img = rng.integers(0, 1024, shape, dtype=cp.uint16)
+    elif int_dtype == cp.int16:
+        # chose a limited range of values to test 512 hist_size case
+        img = rng.integers(-128, 384, shape, dtype=int).astype(cp.int16)
+
+    # 150 is the value used to auto-select between sorting vs. histogram
+    small_kernel = prod(footprint_shape) < 150
+    if algorithm_kwargs and (
+        algorithm == 'sorting'
+        or (algorithm == 'auto' and small_kernel)
+    ):
+        msg = ["algorithm_kwargs={'partitions': 32} ignored"]
+    else:
+        msg = []
+    with expected_warnings(msg):
+        out = median(img, footprint, mode=mode, behavior='ndimage',
+                     algorithm=algorithm, algorithm_kwargs=algorithm_kwargs)
+    expected = ndimage.median_filter(img, size=footprint.shape, mode=mode)
+    assert_allclose(expected, out)
+
+
+# TODO: Determine source of isolated remote test failures when 16-bit range
+#       is > 1024. Could not reproduce locally.
+@pytest.mark.parametrize('mode', ['reflect', ])
+# use an anisotropic footprint large enough to trigger the histogram-based path
+@pytest.mark.parametrize('footprint_shape', [(7, 11)])
+@pytest.mark.parametrize(
+    'int_dtype, irange',
+    [
+        (cp.uint16, (0, 256)),
+        (cp.uint16, (0, 15)),
+        (cp.uint16, (128, 384)),
+        (cp.uint16, (0, 200)),
+        (cp.uint16, (0, 510)),
+        (cp.uint16, (500, 550)),
+        (cp.uint16, (0, 1024)),
+        # (cp.uint16, (0, 2048)),
```

Comment:
Maybe we should mark as `xfail` instead of commenting? Would make it easier to see these issues and fix them later. Also may be worth filing an issue to track. This could be included in the `reason` message

```suggestion
        pytest.param(cp.uint16, (0, 2048), marks=pytest.mark.xfail(reason="<insert reason here>")),
```

Reply:
If I put `xfail` then I will not be able to run locally as the failure doesn't occur for me. I have only seen these fail on CI.

Reply:
We could do `mark.skip` instead

Reply:
Okay this is now using `mark.skip`! Thanks for pointing out the notation to do this viia `pytest.param`.

Reply:
I guess with xfail it doesn't actually cause the tests to fail if they pass, it just gets reported as XPASS in the summary. I think skip is fine, though as we ideally don't expect it to fail.

Reply:
Great thanks Greg! 🙏